### ISSUE TITLE
Just use huggingface

### ISSUE THREAD

**Issue by aeryncaen** [shawwn/llama-dl#6]
All of the models are on huggingface already. https://huggingface.co/decapoda-research

there's even an open, working pr to add support to the transformers lib.

**Comment by shawwn** [comment]
Sure, use whatever works. This repo is intended to serve as a point of communication about llama, and also as an extra mirror.

Note that Facebook has been [issuing takedown requests against huggingface llama repositories](https://twitter.com/ClementDelangue/status/1632948540245671936), so those may get knocked offline.

**Comment by loretoparisi** [comment]
> All of the models are on huggingface already. https://huggingface.co/decapoda-research
> 
> there's even an open, working pr to add support to the transformers lib.

It's worth to note that those models files have been converted to be used in the HF library, so if we take the 7B models files [here](https://huggingface.co/decapoda-research/llama-7b-hf/tree/main)

According the authors the model has been infact

> LLaMA-7B converted to work with Transformers/HuggingFace. This is under a special license, please see the LICENSE file for details.

So supposed we want to use model's file in [C++ inference](https://github.com/ggerganov/llama.cpp) here I'm not sure if would work.

**Comment by tljstewart** [comment]
@loretoparisi Ya i'm thinking along the same lines and trying to make sense here. There are 8bit and 4bit quantized, the original and the huggingface versions... I think  [C++ inference](https://github.com/ggerganov/llama.cpp) use the original weights and converts them, to `ggml` format the authors own format and also does the quantization...? 

Can this be confirmed?

Also, I am currently [using ipfs downloading](https://github.com/facebookresearch/llama/pull/73#issuecomment-1456338060) current time is `2d9h42m` for 65B.... as the magnet link in this repo seems to be down as well as huggingface...

Any thoughts on the model formats with C++ or a way to download the weights faster?

**Comment by loretoparisi** [comment]
yes confirmed. You first convert weights to ggml FP16 or FP32, then quantize to 4bit and run inference (cpu only).

**Comment by tljstewart** [comment]
Ah ok, so your suppose to get the original released weights and the C++ code converts it? Also I found an original weight [torrent link](https://files.catbox.moe/o8a7xw.torrent) and its going extremely fast, ETA 3hour for 235GB.

`webtorrent download o8a7xw.torrent`

**Comment by loretoparisi** [comment]
yes this is exactly what I did from the download here.

**Comment by risos8200** [comment]
You can also use https://huggingface.co/huggyllama, works with llama.cpp.
